File: Two Sum/two_sum.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
    seen = {}
    len = nums.length
    for i in 0...len
        compliment = target - nums[i]
        if seen.key?(compliment)
            return [seen[compliment], i]
        else 
            seen[nums[i]] = i
        end 

    end
    return [-1, -1]
end

File: 0001. Two Sum/two_sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> seen;  // The key is the number, and the value is the index
        for (int i = 0; i < nums.size(); i++) {  // Iterate through the array
            int compliment = target - nums[i];  // Calculate the complement
            if (seen.find(compliment) != seen.end()) {  // Check if complement exists in the map
                return {seen[compliment], i};  // Return the indices
            } else {
                seen[nums[i]] = i;  // Store the index of the current number
            }
        }
        return {-1, -1};  // Return [-1, -1] if no solution is found
    }
};
